src/prommis/solid_handling/tests/test_crusher.py

```diff
@@ -0,0 +1,131 @@
+        m.fs.unit.properties_in[0].particle_size_median.fix(80)  # micrometer
+        m.fs.unit.properties_in[0].particle_size_width.fix(1.5)  # dimensionless
+        m.fs.unit.properties_out[0].particle_size_median.fix(58)  # micrometer
+        m.fs.unit.properties_out[0].particle_size_width.fix(1.5)  # dimensionless
+        return m
+
+    @pytest.mark.build
+    @pytest.mark.unit
+    def test_build(self, model):
+        # Assertions related to the build status of the model
+        # More assertions as needed for your model
+        unit = model.fs.unit
+        assert isinstance(
+            unit.work, Var
+        ), "Variable 'work' is not properly initialized as a Pyomo Var."
+        assert isinstance(
+            unit.feed_p80, Expression
+        ), "Expression 'feed_p80' is not properly initialized as a Pyomo Expression."
+        assert isinstance(
+            unit.prod_p80, Expression
+        ), "Expression 'prod_p80' is not properly initialized as a Pyomo Expression."
+        assert isinstance(
+            unit.crush_work_eq, Constraint
+        ), "Constraint 'crush_work_eq' is not properly initialized as a Pyomo Constraint."
+
+        assert number_variables(model.fs.unit) == 59
+        assert number_total_constraints(model.fs.unit) == 41
+        assert number_unused_variables(model.fs.unit) == 0
+
+    @pytest.mark.component
+    def test_structural_issues(self, model):
+        dt = DiagnosticsToolbox(model=model)
+        dt.assert_no_structural_warnings()
+
+    @pytest.mark.solver
+    @pytest.mark.skipif(solver is None, reason="Solver not available")
+    @pytest.mark.component
+    def test_initialize(self, model):
+        initializer = BlockTriangularizationInitializer(constraint_tolerance=2e-5)
+        initializer.initialize(model.fs.unit)
+        assert initializer.summary[model.fs.unit]["status"] == InitializationStatus.Ok
+
+    @pytest.mark.component
+    @pytest.mark.solver
+    def test_solve(self, model):
+        results = solver.solve(model)
+        assert_optimal_termination(results)
+
+    @pytest.mark.component
+    @pytest.mark.solver
+    def test_solution(self, model):
+        assert pytest.approx(
+            value(model.fs.unit.properties_in[0].flow_mass), rel=1e-5
+        ) == value(model.fs.unit.properties_out[0].flow_mass)
+        for j in model.fs.properties_solid.component_list:
+            assert pytest.approx(
+                value(model.fs.unit.properties_in[0].mass_frac_comp[j]), rel=1e-5
+            ) == value(model.fs.unit.properties_out[0].mass_frac_comp[j])
+
+        assert pytest.approx(114.31301, rel=1e-5) == value(
+            model.fs.unit.feed_p80[0]
+        )  # Test feed size expressions.
+        assert pytest.approx(82.87693, rel=1e-5) == value(
+            model.fs.unit.prod_p80[0]
+        )  # Test prod size expressions.
+
+        assert pytest.approx(3915.710575, rel=1e-5) == value(model.fs.unit.work[0])
```

Comment:
You should add a test using the diagnostics toolbox's `assert_no_numerical_warnings` method on the solved model.